src/test/scala/kubeyml/deployment/KubernetesStateYamlSpec.scala

```diff
@@ -64,22 +63,23 @@ class KubernetesStateYamlSpec extends FlatSpec with Matchers with ScalaCheckProp
   "kubernetes sample" must "be generated from definition" in {
     implicit val arbTest = Properties.lowEmptyChance
     forAll { deploymentTestParts: DeploymentTestParts =>
-        import deploymentTestParts._
-        whenever(nonEmptyParts(deploymentTestParts)) {
-          val deployment = deploy.namespace(namespace)
-            .service(serviceName)
-            .withImage(dockerImage)
-            .withProbes(
-              livenessProbe = HttpProbe(HttpGet("/health", 8080, List.empty), initialDelay = 3 seconds, period = 5 seconds),
-              readinessProbe = NoProbe
-            )
-            .replicas(1)
-            .addPorts(List(Port(Some("app"), 8080)))
-            .annotateSpecTemplate(Map(metadataKey -> metadataValue))
-            .env(envName, envValue)
+      import deploymentTestParts._
+      whenever(nonEmptyParts(deploymentTestParts)) {
+        val deployment = deploy
+          .namespace(namespace)
+          .service(serviceName)
+          .withImage(dockerImage)
+          .withProbes(
+            livenessProbe = HttpProbe(HttpGet("/health", 8080, List.empty), initialDelay = 3 seconds, period = 5 seconds),
+            readinessProbe = NoProbe
+          )
+          .replicas(1)
+          .addPorts(List(Port(Some("app"), 8080)))
+          .annotateSpecTemplate(Map(metadataKey -> metadataValue))
+          .env(envName, envValue)
+          .addCommand(Seq("java"))
```

Comment:
Can we have here an arbitrary for the command to test the following:
- Empty list commands
- Single word commands
- Commands with arguments

Reply:
note: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/

Reply:
API wise, we could make the following choices:

- Seq.head goes in command
- Seq.tail goes in arguments

Or we make something like

- addCommand(cmd: String, arguments: List[String])

And the json encoders and decoders should bridge the differences between this and kubernetes schema

then, the sbt keys can be, one for command which should be a NonEmptyString and one for arguments which defaults to an empty list

Reply:
ok a note on the command, it doesn't have to be empty string, we just need to make this rules hold

> When you override the default Entrypoint and Cmd, these rules apply:
> 
> If you do not supply command or args for a Container, the defaults defined in the Docker image are used.
> 
> If you supply a command but no args for a Container, only the supplied command is used. The default EntryPoint and the default Cmd defined in the Docker image are ignored.
> 
> If you supply only args for a Container, the default Entrypoint defined in the Docker image is run with the args that you supplied.
> 
> If you supply a command and args, the default Entrypoint and the default Cmd defined in the Docker image are ignored. Your command is run with your args.
>

Reply:
Preferable approach would be- Option[NonEmptyString]